### PR TITLE
python3.pkgs.sphinxcontrib-sphinxemoji: init at 0.2.0

### PR DESCRIPTION
**Pull request by KAction**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by KAction** [comment]
This is first in big tree of dependencies necessary to build documentation for `attrs` and everything in between.

I want to build documentation for `attrs`, it depends on number of other packages that also have documentation, which in turn require packaging of even bigger amount of packages.

I can submit the all at once, or submit the one by one, as I package them. Which one is preferred?

File: pkgs/development/python-modules/sphinxcontrib-sphinxemoji/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    cp -r ./docs/build/html $out/share/doc/python/$pname
+  '';
+
+  propagatedBuildInputs = [ sphinx ];
+  nativeBuildInputs = [ sphinx ];
+
+  meta = with lib; {
+    description = "extension to use emoji codes in your Sphinx documentation";
+    homepage = "https://github.com/sphinx-contrib/emojicodes";
+    license = licenses.mit;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sphinxcontrib-sphinxemoji/default.nix:29]: Missing maintainer.

**Comment by SuperSandro2000** [comment]
> I can submit the all at once, or submit the one by one, as I package them. Which one is preferred?

Maybe you can combine the dependencies of a package if they are less then 5?

**Comment by onny** [comment]
Please rebase

**Comment by KAction** [comment]
Rebased

File: pkgs/development/python-modules/sphinxemoji/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    cp -r ./docs/build/html $out/share/doc/python/$pname
+  '';
+
+  propagatedBuildInputs = [ sphinx ];
+  nativeBuildInputs = [ sphinx ];
+
+  meta = with lib; {
+    description = "extension to use emoji codes in your Sphinx documentation";
+    homepage = "https://github.com/sphinx-contrib/emojicodes";
+    license = licenses.mit;
```
Comment (onny) [inline pkgs/development/python-modules/sphinxemoji/default.nix:29]:
```suggestion
    license = licenses.mit;
    maintainers = with maintainers; [ kaction ];
```
if you like :) but maintainer should be added

File: pkgs/development/python-modules/sphinxemoji/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  postBuild = ''
+    PYTHONPATH=$PWD:$PYTHONPATH make -C docs html
+  '';
+  postInstall = ''
+    mkdir -p $out/share/doc/python/$pname
+    cp -r ./docs/build/html $out/share/doc/python/$pname
+  '';
+
+  propagatedBuildInputs = [ sphinx ];
+  nativeBuildInputs = [ sphinx ];
```
Comment (onny) [inline pkgs/development/python-modules/sphinxemoji/default.nix:24]:
```suggestion
  propagatedBuildInputs = [ sphinx ];
  nativeBuildInputs = [ sphinx ];
```
I'm more familiar with putting these two after `src`.

File: pkgs/development/python-modules/sphinxemoji/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  version = "0.2.0";
+  outputs = [ "out" "doc" ];
+
+  src = fetchFromGitHub {
+    owner = "sphinx-contrib";
+    repo = "emojicodes"; # does not match pypi name
+    rev = "v${version}";
+    sha256 = "sha256-TLhjpJpUIoDAe3RZ/7sjTgdW+5s7OpMEd1/w0NyCQ3A=";
+  };
+  pythonImportsCheck = [ "sphinxemoji" ];
```
Comment (onny) [inline pkgs/development/python-modules/sphinxemoji/default.nix:14]:
```suggestion
  pythonImportsCheck = [ "sphinxemoji" ];
```
I'm more familiar with putting this line after `postInstall`

**Review by onny — COMMENTED** [review]
Otherwise LGTM :)

**Comment by KAction** [comment]
Applied and rebased.

File: pkgs/development/python-modules/sphinxemoji/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  postInstall = ''
+    mkdir -p $out/share/doc/python/$pname
+    cp -r ./docs/build/html $out/share/doc/python/$pname
+  '';
+
+  pythonImportsCheck = [ "sphinxemoji" ];
+
+  meta = with lib; {
+    description = "extension to use emoji codes in your Sphinx documentation";
```
Comment (onny) [inline pkgs/development/python-modules/sphinxemoji/default.nix:32]:
```suggestion
    description = "Extension to use emoji codes in your Sphinx documentation";
```

Comment (KAction) [reply pkgs/development/python-modules/sphinxemoji/default.nix]: Okay.

**Comment by onny** [comment]
Result of `nixpkgs-review pr 145443` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.sphinxemoji</li>
    <li>python39Packages.sphinxemoji</li>
  </ul>
</details>

**Comment by onny** [comment]
Thank you!

**Pull request merged**